react/src/components/AutoScalingRuleEditorModal.tsx

```diff
@@ -0,0 +1,320 @@
+              comparator: values.comparator,
+              step_size: values.step_size,
+              cooldown_seconds: values.cooldown_seconds,
+              min_replicas: values.min_replicas,
+              max_replicas: values.max_replicas,
+            },
+          },
+          onCompleted: (res, errors) => {
+            message.success(t('autoScalingRule.SuccessfullyCreated'));
+            onRequestClose(true);
+          },
+          onError: (error) => {
+            message.error(error.message);
+            onRequestClose(false);
+          },
+        });
+      }
+    });
+  };
+
+  const handleCancel = () => {
+    onRequestClose();
+  };
+
+  const comparators = [
+    { label: '<', value: 'LESS_THAN' },
+    { label: '<=', value: 'LESS_THAN_OR_EQUAL' },
+    { label: '>', value: 'GREATER_THAN' },
+    { label: '>=', value: 'GREATER_THAN_OR_EQUAL' },
+  ];
+
+  return (
+    <BAIModal
+      {...baiModalProps}
+      destroyOnClose
+      onOk={handleOk}
+      onCancel={handleCancel}
+      centered
+      title={
+        autoScalingRule
+          ? t('autoScalingRule.EditAutoScalingRule')
+          : t('autoScalingRule.AddAutoScalingRule')
+      }
+      confirmLoading={
+        isInflightAddAutoScalingRule || isInflightModifyAutoScalingRule
+      }
+    >
+      <Form
+        ref={formRef}
+        layout={'vertical'}
+        requiredMark={'optional'}
+        initialValues={
+          autoScalingRule
+            ? autoScalingRule
+            : {
+                metric_source: 'KERNEL',
+                comparators: 'LESS_THAN',
+                threshold: '10',
+                step_size: 0,
+                cooldown_seconds: 300,
+                min_replicas: 0,
+                max_replicas: 3,
+              }
+        }
+      >
+        <Form.Item
+          label={t('autoScalingRule.MetricName')}
+          name={'metric_name'}
+          rules={[{ required: true }]}
+        >
+          <Input />
+        </Form.Item>
+        <Form.Item
+          label={t('autoScalingRule.MetricSource')}
+          name={'metric_source'}
+          rules={[{ required: true }]}
+        >
+          <Select>
+            <Select.Option value={'INFERENCE_FRAMEWORK'}>
+              Inference Framework
+            </Select.Option>
+            <Select.Option value={'KERNEL'}>Kernel</Select.Option>
+          </Select>
+        </Form.Item>
+        <Flex
+          direction="row"
+          justify="around"
+          align="stretch"
+          style={{ width: '100%' }}
+          gap={'md'}
+        >
+          <Form.Item
+            label={t('autoScalingRule.Comparator')}
+            name={'comparator'}
+            rules={[{ required: true }]}
+            style={{ width: 600 }}
+          >
+            <Radio.Group
+              block
+              size={'small'}
+              options={comparators}
+              optionType={'button'}
+              buttonStyle={'solid'}
+            ></Radio.Group>
```

Comment:
Please do not use solid button style yet. It'll be applied globally. :)
```suggestion
            ></Radio.Group>
```